src/com/minseungkwon/datastructure/Graph/Graph.java

```diff
@@ -0,0 +1,119 @@
+package com.minseungkwon.datastructure.Graph;
+
+import com.minseungkwon.datastructure.Queue;
+
+import java.util.LinkedList;
+import java.util.Stack;
+
+public class Graph {
+
+    class Node {
+        int data;
+        LinkedList<Node> adjacent;
+        boolean marked;
+
+        Node(int data) {
+            this.data = data;
+            adjacent = new LinkedList<>();
+            marked = false;
```

Comment:
What happens if we call DFS more than once on the same Graph Object? I think this marked state should be managed inside DFS or DFS Recursive function itself.